dist/comment_style.py

```diff
@@ -0,0 +1,148 @@
+                # preceding the comment in subsequent lines.
+                if length != 0 and not text_check.match(line.strip()):
+                    # If the comment is length 2 and we found another matching line then we have
+                    # found an illegal comment.
+                    if length == 2:
+                        print_msg(file_name, line_num, line, "multiline ")
+                        count += 1
+                    length += 1
+                    # Try and print only one error per comment, just keep incrementing the count and
+                    # the above if will only run once.
+                else:
+                    length += 1
+            else:
+                length = 0
+            line_num += 1
+        return count
+
+def file_is_cpp(name):
+    if re.search('(.cpp|.cxx|.hpp)$', name) is not None:
+        return True
+    if re.search('(.c|.i|.in)$', name) is not None:
+        return False
+
+    # House style is that C++ header files use ".h", which unfortunately makes
+    # this sort of code rather difficult. Luckily, libmagic can identify C/C++
+    # based on content. Don't import it because Python packaging is a disaster
+    # and this script needs to run reliably.
+    result = subprocess.run("file {}".format(name),
+                            shell=True,
+                            capture_output=True,
+                            text=True).stdout.strip('\n')
+    return "C++" in result
+
+# A collection of cases we handle.
+tests = [
+    (r'// hello world', True),
+    (r'     // hello world', True),
+    (r'hello world', False),
+    (r'printf("Hello, World!\n"); // prints hello world', True),
+    (r'String url = "http://www.example.com"', False),
+    (r'// hello world', True),
+    (r'//\\', True),
+    (r'// "some comment"', True),
+    (r'new URI("http://www.google.com")', False),
+    (r'printf("Escaped quote\""); // Comment', True),
+    (r' * http://www.google.com', False)
+]
+
+def validate(line, expect_match):
+    if (expect_match and not match(line.strip())) or (not expect_match and match(line.strip())):
+        print("Test failed:" + line)
+    else:
+        print("Test success")
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-t', '--test', default=False,
+                        help='Run self tests', action='store_true')
+    parser.add_argument('-f', '--fast', default=False,
+                        help='Use fast mode by only checking files changed in git',
+                        action='store_true')
+    args = parser.parse_args()
+
+    if args.test:
+        for test in tests:
+            validate(test[0], test[1])
+            sys.exit(0)
+
+    # Move up to root dir.
+    os.chdir("..")
+
+    # Some directories aren't expected to comply with WiredTiger style. Ignore them.
+    ignore_directories=[
+        'src/checksum',
```

Comment:
When I originally wrote this I think i found a number of false positives in these directories and ignored them. I tried un-ignoring them and running the script and there is a tiny number of failures. We should remove this array and fix those failures if we can.

Reply:
Ack, I'll give it a shot.

Reply:
Done, `mtx_rw.c` had a bunch of embedded comments so I've left that in the list and fixed the others.